From fdef5b662df941c49fd2d6e486d0427f8dfb602e Mon Sep 17 00:00:00 2001
From: Jose Quaresma <quaresma.jose@gmail.com>
Date: Sun, 7 Feb 2021 01:30:39 +0000
Subject: [PATCH] cmake: disable building external dependencies in /third_party

Upstream-Status: Inappropriate [OE-core specific]

Signed-off-by: Jose Quaresma <quaresma.jose@gmail.com>

---
 CMakeLists.txt       | 3 ---
 glslc/CMakeLists.txt | 1 -
 2 files changed, 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0c8d1ae..c7940f6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -111,9 +111,6 @@ endif(MSVC)
 
 
 # Configure subdirectories.
-# We depend on these for later projects, so they should come first.
-add_subdirectory(third_party)
-
 add_subdirectory(libshaderc_util)
 add_subdirectory(libshaderc)
 add_subdirectory(glslc)
diff --git a/glslc/CMakeLists.txt b/glslc/CMakeLists.txt
index 31664d1..358d91b 100644
--- a/glslc/CMakeLists.txt
+++ b/glslc/CMakeLists.txt
@@ -53,7 +53,6 @@ shaderc_default_compile_options(glslc_exe)
 target_include_directories(glslc_exe PRIVATE ${CMAKE_CURRENT_BINARY_DIR}/.. ${spirv-tools_SOURCE_DIR}/include)
 set_target_properties(glslc_exe PROPERTIES OUTPUT_NAME glslc)
 target_link_libraries(glslc_exe PRIVATE glslc shaderc_util shaderc)
-add_dependencies(glslc_exe build-version)
 
 shaderc_add_tests(
   TEST_PREFIX glslc
